atomicapp/providers/openshift.py

```diff
@@ -133,6 +146,49 @@ def process_template(self, url, template):
             logger.error(msg)
             raise ProviderFailedException(msg)
 
+    def execute(self, namespace, pod, container, command):
+        """
+        Execute a command in a container in an Openshift pod.
+
+        Args:
+            namespace (str): Namespace
+            pod (str): Pod name
+            container (str): Container name inside pod
+            command (str): Command to execute
+
+        Returns:
+            Command output (str)
+        """
+        args = {
+            'token': self.access_token,
+            'namespace': namespace,
+            'pod': pod,
+            'container': container,
+            'command': ''.join(['command={}&'.format(word) for word in command.split()])
+        }
+        url = 'wss://' + urljoin(
+            self.kubernetes_api,
+            'namespaces/{namespace}/pods/{pod}/exec?'
+            'access_token={token}&container={container}&'
+            '{command}stdout=1&stdin=0&tty=0'.format(
+              **args)).split('://', 1)[-1]
+
+        results = []
+
```

Comment:
Probably need to comment on why we needed websockets here.

Basic https won't work because the request needed to be upgraded to SPDY, which python-requests doesn't support. The endpoint does work with websockets though, so we had to take this route.